commcare_connect/program/helpers.py

```diff
@@ -130,3 +131,16 @@ def get_delivery_performance_report(program: Program, start_date, end_date):
     )
 
     return managed_opportunities
+
+
+@quickcache(vary_on=["current_org_slug", "opp_id"], timeout=24 * 60 * 60)
+def is_org_manages_opportunity_via_program(current_org_slug, opp_id: int):
+    """
+    Check if the current organization manages the given managed opportunity
+    through any of its programs.
+    """
+    programs = Program.objects.filter(organization__slug=current_org_slug)
+    if programs:
+        return ManagedOpportunity.objects.filter(program__in=programs, id=opp_id).exists()
+
+    return False
```

Comment:
This can be simplified to just a single query to check if the managed opportunity exists through a program managed by current organization.
```suggestion
    return ManagedOpportunity.objects.filter(program__organization__slug=current_org_slug, id=opp_id).exists()
```

Reply:
Thanks - [8b6514f](https://github.com/dimagi/commcare-connect/pull/533/commits/8b6514f19ed01aff9e82ea3054a9f5e367d0ba25)